From 34996ad1eb6d295382e7737ef34bd362abdd4944 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 19 Jan 2021 22:15:45 +0100
Subject: [PATCH] photosyst.c: fix build on musl
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Build on musl is broken since version 2.6.0 and
https://github.com/Atoptool/atop/commit/e889c66fbe1d0b7ae38fbcbaa46cea749257f486
because limits.h is not included:

photosyst.c: In function ‘lvmmapname’:
photosyst.c:1624:14: error: ‘PATH_MAX’ undeclared (first use in this function); did you mean ‘AF_MAX’?
   char  path[PATH_MAX];
              ^~~~~~~~
              AF_MAX

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 photosyst.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/photosyst.c b/photosyst.c
index 3de874b..67ae92f 100644
--- a/photosyst.c
+++ b/photosyst.c
@@ -165,6 +165,7 @@
 #include <dirent.h>
 #include <sys/ioctl.h>
 #include <sys/sysmacros.h>
+#include <limits.h>
 
 #define SCALINGMAXCPU	8	// threshold for scaling info per CPU
 
-- 
2.29.2

